test/fail_compilation/test14538.d

```diff
@@ -0,0 +1,20 @@
+// PERMUTE_ARGS:
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test14538.d(19): Error: cannot implicitly convert expression (x >= 0 ? cast(uint)this.fCells[cast(ulong)x].code : 32u) of type uint to Cell
+---
+*/
+
+struct Cell
+{
+    dchar code;
+    alias code this;
+}
+
+struct Row
+{
+    Cell[] fCells;
+    Cell opIndex(int x) { return (x >= 0) ? fCells[x] : ' '; }
```

Comment:
s/`int x`/`size_t x`/ will remove the difference of diagnostic message between 32/64bit platforms.